icu4c/source/i18n/number_longnames.cpp

```diff
@@ -350,7 +547,7 @@ UnicodeString LongNameHandler::getUnitDisplayName(
         return ICU_Utility::makeBogusString();
     }
     UnicodeString simpleFormats[ARRAY_LENGTH];
-    getMeasureData(loc, unit, width, simpleFormats, status);
+    getMeasureData(loc, unit, width, "", simpleFormats, status);
```

Comment:
discuss this offline

Reply:
Happy to discuss!
(As to this parameter: display names aren't inflected.)